src/Database/Query.php

```diff
@@ -277,12 +309,12 @@ public function sql(?ValueBinder $generator = null): string
      * ```
      *
      * @param callable $visitor A function or callable to be executed for each part
-     * @param array $parts The query clauses to traverse
      * @return $this
      */
-    public function traverse(callable $visitor, array $parts = [])
+    public function traverse(callable $visitor)
     {
-        $parts = $parts ?: array_keys($this->_parts);
+        $partList = "_{$this->type()}Parts";
+        $parts = $this->{$partList} ?? array_keys($this->_parts);
```

Comment:
I would really like to keep the parts in the query compiler, unless you think this is a better idea.

One way to keep that design is by adding a traverseParts() function to the query, with the signature and implementation this function had before.

Reply:
I am not convinced what I have is great. Adding a `traverseParts` method to Query sounds good to me.